src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -135,25 +135,33 @@
         this.__subSubContext__[obj.daoKey].find(obj.objId).then(requestObj => {
           let referenceSummaryString = `ID:${obj.objId}`;
 
-          if ( requestObj && requestObj.toSummary() ){
-            referenceSummaryString = requestObj.toSummary()
-          }
+          if ( requestObj ){
+            Promise.resolve(requestObj.toSummary()).then(function(requestObjSummary) {
```

Comment:
If we make toSummary() able to return Promises, then we need to change everywhere in the system that we use toSummary(), not just here. FYI: @jlhughes

Reply:
Think only place so far where it returns a promise is for `Transaction.js`, let me just double check if we actually need that Promise.resolve()

Reply:
Confirmed `Transaction.toSummary()` returns a promise,

**TableCellFormatter** seems to magically handle the `Transaction.toSummary(`) and render its resolved Promise without having to directly resolve it, I’m guessing `Element.add` can take promises

But yeah for **View**, we just set a simple slot to `Transaction.toSummary()`, so its setting the slot as the actual Promise and it doesn’t render anything if you don’t resolve it.

Reply:
Left TODO for ReferenceView